src/fluree/db/json_ld/policy/rules.cljc

```diff
@@ -78,22 +78,25 @@
   (map? target-expr))
 
 (defn parse-targets
-  [db fuel-tracker policy-values target-exprs]
+  [db fuel-tracker error-ch policy-values target-exprs]
   (let [in-ch  (async/to-chan! target-exprs)
         out-ch (async/chan 2 (map (fn [iri] (iri/iri->sid iri (:namespaces db)))))]
     (async/pipeline-async 2
                           out-ch
                           (fn [target-expr ch]
-                            (if (query-target? target-expr)
-                              (let [context (json-ld/parse-context (get target-expr "@context"))
-                                    sid-xf  (map #(json-ld/expand-iri % context))
-                                    target-q (cond-> (assoc target-expr "select" "?$target")
-                                               policy-values (policy/inject-where-pattern ["values" policy-values]))]
-                                (-> (dbproto/-query db fuel-tracker target-q)
-                                    (async/pipe (async/chan 2 (comp cat sid-xf)))
-                                    (async/pipe ch)))
-                              ;; non-maps are literals
-                              (async/onto-chan! ch [target-expr])))
+                            (async/go
+                              (try*
+                                (if (query-target? target-expr)
+                                  (let [target-q (cond-> (assoc target-expr
+                                                                "select" "?$target"
+                                                                :selection-context {}) ;; don't compact selection results
```

Comment:
What is `:selection-context` for?

Reply:
This was introduced with other rules parsing.

It provides a separate context for selection results, different than the context used in the where clause.

The goal is the same as done here - not have to compact the select result, then expand it again - which is expensive.

Reply:
That's a useful feature, I like it. We may want to call it something more generic, like `result-context`, since we use that context for `:construct` results in addition to regular `:select`/`:select-distinct`/`:select-one` results.

Reply:
I'd support that name change. It isn't documented at this point, just used internally in similar situations. I think outside the scope here as that would impact other non-policy code.